### PR TITLE
Generate init/clear functions for TreeIter, TextIter and Border

### PR DESCRIPTION
**Pull request by sdroege**
See https://github.com/gtk-rs/gir/pull/793

----

Needs https://github.com/gtk-rs/glib/pull/496 and the gir PR merged first.

### CODE REVIEW

**Comment by sdroege** [comment]
CC @EPashkin @GuillaumeGomez

**Comment by EPashkin** [comment]
Thanks, 👍

File: Gir.toml [hunk 1]
```diff
@@ -1781,6 +1780,9 @@ manual_traits = ["TextBufferExtManual"]
 [[object]]
 name = "Gtk.TextIter"
 status = "generate"
+init_function_expression = "|_ptr| ()"
```
Comment (GuillaumeGomez) [inline Gir.toml]: That is really super ugly... Maybe we should think about an enum?

**Comment by GuillaumeGomez** [comment]
There are auto code differences.

**Comment by sdroege** [comment]
> There are auto code differences.

What do you mean?

File: Gir.toml (hunk 1 above)
Comment (sdroege) [reply Gir.toml]: How would that enum look like? We somehow need to provide a way to provide a closure here as there are too many different cases.

**Comment by GuillaumeGomez** [comment]
The CI error.

**Comment by sdroege** [comment]
> The CI error.

Yes, that needs the glib PR merged first. Then it succeeds.

**Comment by EPashkin** [comment]
@sdroege You forgot to update gir submodule too

**Comment by sdroege** [comment]
> @sdroege You forgot to update gir submodule too

I was waiting with the final update for the GLib PR to be merged so that the CI can succeed then.

**Comment by GuillaumeGomez** [comment]
Ah sorry, thought that the gir PR was merged!

**Comment by sdroege** [comment]
> Ah sorry, thought that the gir PR was merged!

The gir PR was, the GLib one not. We need the GLib one for the CI to succeed (and I'll have to update this one, the GTK PR, for the updated gir submodule).

**Comment by GuillaumeGomez** [comment]
As I said on IRC: I don't have time to investigate on how to improve the syntax of this feature for the moment, therefore I'll open a PR and come back to it later if I'm able to find something more "user friendly". The linked issue is here: https://github.com/gtk-rs/gir/issues/794

Thanks @sdroege!

**Comment by sdroege** [comment]
@GuillaumeGomez Good to go :)

**Comment by GuillaumeGomez** [comment]
Thanks!

**Pull request merged**